src/pages/LogOutPreviousUserPage.tsx

```diff
@@ -1,79 +1,67 @@
-import lodashGet from 'lodash/get';
-import PropTypes from 'prop-types';
+import {RouteProp} from '@react-navigation/native';
 import React, {useEffect} from 'react';
 import {Linking} from 'react-native';
-import {withOnyx} from 'react-native-onyx';
+import {OnyxEntry, withOnyx} from 'react-native-onyx';
 import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
 import * as SessionUtils from '@libs/SessionUtils';
 import Navigation from '@navigation/Navigation';
-import * as Session from '@userActions/Session';
+import * as SessionUserAction from '@userActions/Session';
 import ONYXKEYS from '@src/ONYXKEYS';
-import ROUTES from '@src/ROUTES';
+import ROUTES, {Route} from '@src/ROUTES';
+import type {Account, Session} from '@src/types/onyx';
 
-const propTypes = {
+type LogOutPreviousUserPageOnyxProps = {
     /** The details about the account that the user is signing in with */
-    account: PropTypes.shape({
-        /** Whether the account data is loading */
-        isLoading: PropTypes.bool,
-    }),
+    account: OnyxEntry<Account>;
 
     /** The data about the current session which will be set once the user is authenticated and we return to this component as an AuthScreen */
-    session: PropTypes.shape({
-        /** The user's email for the current session */
-        email: PropTypes.string,
-    }),
+    session: OnyxEntry<Session>;
 };
 
-const defaultProps = {
-    account: {
-        isLoading: false,
-    },
-    session: {
-        email: null,
-    },
+type LogOutPreviousUserPageProps = LogOutPreviousUserPageOnyxProps & {
+    route: RouteProp<{params: {shouldForceLogin?: string; email?: string; shortLivedAuthToken?: string; exitTo?: Route}}>;
 };
 
-function LogOutPreviousUserPage(props) {
+function LogOutPreviousUserPage({account = {isLoading: false}, session, route}: LogOutPreviousUserPageProps) {
```

Comment:
What about a destruction assignment for this case, because this chain `route.params` is being used often.
```suggestion
function LogOutPreviousUserPage({account = {isLoading: false}, session, route: {params}}: LogOutPreviousUserPageProps) {
```